### DISCUSSION TITLE
Replacing every instance in the whole document

### DISCUSSION THREAD

**Discussion by avtfaciam** [jglev/obsidian-apply-patterns-plugin#20] [Q&A] [answered]
Hi! Thank you so much for this plugin, it seems to be exactly what I've been looking for. I wanted some way to apply 4 different "search and replace" commands quickly and this plugin seemed to fix that.

However, I don't know if it's a setting I unwittingly toggled or not, but every time I run the command it only replaces _one instance_ of the pattern. 

Here's my pattern for reference:
<img width="633" alt="Screen Shot 2021-09-01 at 01 31 47" src="https://user-images.githubusercontent.com/66343542/131612233-ee844000-a495-4e27-b290-2dc0f2ae6522.png">

So in a text, it would only replace the first occurrence of each pattern, instead of replacing all occurrences of all patterns.

Could you please guide me the right way? Thanks!

**Comment by FelipeRearden** [comment, marked as answer]
@enjolrs 

Turn the `g` (Which means Global) option ON to every rule and try again.  It works for me :)

**Reply by jglev** [reply]
Thank you, @FelipeRearden! I agree : ) @enjolrs, does that work for you?

**Reply by jglev** [reply]
Also, @enjolrs, thank you for your kind words!

**Reply by FelipeRearden** [reply]
I’m glad to help @publicus !!!!

**Reply by avtfaciam** [reply]
It worked, thank you!!
